fastlane/lib/fastlane/fast_file.rb

```diff
@@ -338,7 +338,15 @@ def import_from_git(url: nil, branch: 'HEAD', path: 'fastlane/Fastfile', version
             UI.user_error!("No tag found matching #{version.inspect}") if checkout_param.nil?
           end
 
-          Actions.sh("cd #{clone_folder.shellescape} && git checkout #{checkout_param.shellescape} #{checkout_path}")
+          if is_eligible_for_caching && version.nil?
+            # Update the repo if it's eligible for caching but the version isn't specified
+            UI.message("Fetching remote git branches and updating git repo...")
+            Helper.with_env_values('GIT_TERMINAL_PROMPT' => '0') do
+              Actions.sh("cd #{clone_folder.shellescape} && git fetch --all -q && git checkout #{checkout_param.shellescape} #{checkout_path} && git reset --hard && git merge")
```

Comment:
I feel like a `rebase` would make more sense than a `merge`. What do you say?

Reply:
Sounds like a good idea.

To be clear, the only difference between `git merge` and `git rebase` is whether to create a merge commit or not if there are additional commits that a remote branch doesn’t have. Even though it’s not a preferable thing (e.g. adding commits directly from the repo in the cache path), but I think not creating a merge commit would be better in this case.

Reply:
Exactly 👍🏻